graphics/src/ColladaExporter.cc

```diff
@@ -239,10 +242,75 @@ void ColladaExporter::Export(const Mesh *_mesh, const std::string &_filename,
     colladaXml->LinkEndChild(libraryEffectsXml);
   }
 
+  tinyxml2::XMLElement *libraryLightsXml =
+      xmlDoc.NewElement("library_lights");
+  for (const auto& light : _lights)
+  {
+    tinyxml2::XMLElement *lightXml =
+      xmlDoc.NewElement("light");
+    lightXml->SetAttribute("name", light.name.c_str());
+    lightXml->SetAttribute("id", light.name.c_str());
+    libraryLightsXml->LinkEndChild(lightXml);
+
+    tinyxml2::XMLElement *techniqueCommonXml =
+      xmlDoc.NewElement("technique_common");
+    lightXml->LinkEndChild(techniqueCommonXml);
+
+    tinyxml2::XMLElement *lightTypeXml =
+      xmlDoc.NewElement(light.type.c_str());
+    techniqueCommonXml->LinkEndChild(lightTypeXml);
+
+    // color
+    tinyxml2::XMLElement *colorXml =
+      xmlDoc.NewElement("color");
+    char color_str[100] = { 0 };
+    snprintf(color_str, sizeof(color_str), "%g %g %g",
+      light.diffuse.R(), light.diffuse.G(), light.diffuse.B());
```

Comment:
can we use `std::string` instead of `char*`. Same below

Reply:
there's a couple of reasons why i use non-c++ methods here:
1) The syntax is a lot more readable; the c++ style is a lot more cumbersome
2) There are other areas in this file where we use snprintf; I'd rather we stick to the snprintf styles to maintain consistency and get the benefits of (1)